serve/mlc_serve/model/paged_cache_model.py

```diff
@@ -241,7 +241,18 @@ def _apply_top_p_top_k(logits, top_ps, top_ks):
     return logits
 
 
-def sample(logits, sampling_params, vocab_size):
+def sample(
+    logits: Union[tvm.nd.NDArray, torch.Tensor],
+    sampling_params: List[SamplingParams],
+    vocab_size: int,
+    check_safety=False,
+) -> Optional[np.ndarray]:
+    def _is_safe_to_sample(prob_like):
+        return (
+            torch.sum(torch.isnan(prob_like) | torch.isinf(prob_like) | (prob_like < 0))
```

Comment:
Did you evaluate performance implication by adding of this verification?

Reply:
We don't run this check in a normal path. `check_safety` is False by default and set to True only after the exception is raised.